src/broker/cas_cgw.c

```diff
@@ -2438,3 +2456,279 @@ cgw_utype_to_string (int type)
       return (char *) "";
     }
 }
+
+char *
+cgw_rewrite_query (char *src_query)
+{
+  char *rewrite_query = NULL;
+  char *select_from = NULL;
+  char *inline_view = NULL;
+  char *new_inline_view = NULL;
+  char *cublink = NULL;
+  char *cols_type = NULL;
+  char **cols_list = NULL;
+  char *where = NULL;
+
+  char *start = NULL;
+  char *end = NULL;
+
+  size_t str_size = 0;
+  SQLSMALLINT num_cols = 0;
+  bool is_rewrite_error = false;
+  int err_code = 0;
+
+  char col_name[COL_NAME_LEN + 1];
+  SQLSMALLINT col_name_length = 0;
+  SQLHSTMT hstmt;
+
+  if (strstr (src_query, REWRITE_DELIMITER_CUBLINK) == NULL)
+    {
+      is_rewrite_error = true;
+      goto REWRITE_ERROR;
+    }
+
+  start = strstr (src_query, REWRITE_DELIMITER_FROM);
+  if (start == NULL)
+    {
+      is_rewrite_error = true;
+      goto REWRITE_ERROR;
+    }
+
+  str_size = ((start + strlen (REWRITE_DELIMITER_FROM)) - src_query);
```

Comment:
it's better to define a macro for "length of REWRITE_DELIMITER_FROM" as a constant.

Reply:
fixed it